composer.json

```diff
@@ -226,22 +226,21 @@
         }
     },
     "require": {
-        "php": ">=5.3.0"
-    },
-    "autoload": {
-        "psr-4": {
-            "ccxt\\": "php"
-        }
+        "php": ">=5.3.0",
+        "ext-curl": "*",
+        "ext-iconv": "*",
+        "ext-pcre": "*",
+        "symfony/polyfill-mbstring": "^1.7"
```

Comment:
I'm not sure about this particular way, as far as my experience with PHP goes, there's usually a way to use a built mbstring extensions as any other extensions listed above, is there a need for the polyfill? The major concern with the kind of libraries like ccxt is the security, so, less dependencies is better, and we would prefer the *native* shipped dependencies to 3rd-party polyfills. Can you please clarify this? I also wonder if it breaks class autoloading like this...

Reply:
Some hosted php environments, don't ship with mbstring, so if this happens the polyfill-mbstring is used else the php-mbstring. And why should the autoload break?

Reply:
Don't know, I was mostly asking you ) Merging it anyway shortly )

Reply:
No it wont break any autoload in php, and i removed the second autoload from composer.json (composer is happy now :dancer: )

Reply:
Removing autoload section will likely break autoloading. There is a old way of requiring ccxt.php, but letting composer do it's job is a better practice IMO.

Reply:
this is what i mean, this file had 2 of them

Reply:
> There is a old way of requiring ccxt.php, but letting composer do it's job is a better practice IMO.

The same way of loading them is used in PHP exchanges, can you show how to let composer do it's job and how I should link against the lib the modern way?

Reply:
simple example:

```php
<?php
require_once 'vendor/autoload.php';

use ccxt\acx

$class = new acx(array('option' => 'test'));
$class->function;
```

Reply:
@prisis but we don't have a vendor/autoload.php file in the repo...

Reply:
Anyway, nvm, I think I just need to study Composer more...

Reply:
Contents of `vendor/` dir are autogenerated by composer. So, once someone calls:

`composer install`

within repository root (actually, any directory containing `composer.json` file) - all of that stuff will be generated / dependencies will be downloaded, etc.

Or, if someone requires ccxt as a dependancy, then composer will download it into their `vendor/` dir and based on `autoload` section will prepare a proper autoloader for the lib.

I'll check examples tomorrow and send PRs if required.

Reply:
@kornrunner , @prisis thx for explaining it! And respect to both of you for your involvement!